src/DependencyInjection/Configuration.php

```diff
@@ -40,6 +40,11 @@ public function getConfigTreeBuilder()
                             ->scalarNode('port')->end()
                             ->scalarNode('user')->end()
                             ->scalarNode('pwd')->end()
+                            ->booleanNode('ssl')->defaultFalse()->end()
+                            ->arrayNode('ssl_options')
+                                ->info('Define any options from SSL context options. See https://www.php.net/manual/context.ssl.php')
+                                ->variablePrototype()->end()
```

Comment:
I didn't find any doc about `variablePrototype` method.
What is its purpose ?

Reply:
`variablePrototype`  allow any elements of the `ssl_options` array to have any type (or, to mimic the code: `to have a variable type`). In this case I wanted to avoid hardcoding all config exposed by the SSL context options and some can be fed with `array` ;-)

Reply:
On ne peut pas ajouter des [`children`](https://symfony.com/doc/3.3/bundles/configuration.html#processing-the-configs-array) pour préciser la configuration attendues dans le tableau `ssl_options` ?

Ça va peut-être à l'encontre de ce que tu précisais (auquel cas, je n'ai pas compris ta remarque) :

> In this case I wanted to avoid hardcoding all config exposed by the SSL context options

Reply:
Si si @fhuitelec, on peut, mais, pour moi, ce n’est pas souhaitable, ça nous lie à la configuration du contexte SSL car [la configuration est utilisées telle quelle dans php-amqplib](https://github.com/php-amqplib/php-amqplib/blob/cfbfaf6262cd8d017f29862164f75e265d832434/PhpAmqpLib/Connection/AMQPSSLConnection.php#L58-L63).

C’était exactement le sens de ma remarque : je ne veux pas me lier la config du bundle au contexte SSL de PHP.